nodejs/aws-infra/experimental/ec2.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "../utils";
+
+import * as mod from ".";
+
+export type EC2TaskDefinitionArgs = utils.Overwrite<mod.TaskDefinitionArgs, {
+    /** Not provided.  Defaults automatically to ["EC2"] */
+    requiresCompatibilities?: never;
+
+    /**
+     * Not provided for ec2 task definitions.
+     */
+    cpu?: never;
+    /**
+     * Not provided for ec2 task definitions.
+     */
+    memory?: never;
+
+    /**
+     * Single container to make a ClusterTaskDefinition from.  Useful for simple cases where there
+     * aren't multiple containers, especially when creating a ClusterTaskDefinition to call [run]
+     * on.
+     *
+     * Either [container] or [containers] must be provided.
+     */
+    container?: mod.ContainerDefinition;
+
+    /**
+     * All the containers to make a ClusterTaskDefinition from.  Useful when creating a
+     * ClusterService that will contain many containers within.
+     *
+     * Either [container] or [containers] must be provided.
+     */
+    containers?: Record<string, mod.ContainerDefinition>;
+}>;
+
+export class EC2TaskDefinition extends mod.TaskDefinition {
```

Comment:
Is this standard casing for TypeScript, versus, say, `Ec2TaskDefinition`? (Really just asking.)

Reply:
It's incredibly over the map, primarily because TS just wraps JS, and JS does wahtever it wants per library.